File: fourth/php/libreria/licenses.php
<?php
abstract class Licenses{
    private int $cost;
    private int $duration;
    private string $description;
    public function __construct(int $cost, int $duration, string $description) {
        $this->cost = $cost;
        $this->duration = $duration;
        $this->description = $description;
    }
    public function getCost(): int {
        return $this->cost;
    }
    public function getDuration(): int {
        return $this->duration;
    }
    public function getDescription(): string {
        return $this->description;
    }
    abstract public function calculate(): float;
}
?>

File: fourth/java/Licenses.java
abstract public class Licenses {
    private int cost;
    private int duration;
    private String description;

    public Licenses(int cost, int duration, String description) {
        this.cost = cost;
        this.duration = duration;
        this.description = description;
    }

    public int getCost() {
        return this.cost;
    }

    public int getDuration() {
        return this.duration;
    }

    public String getDescription() {
        return this.description;
    }

    public abstract float calculate();
}
